metrics-scraper/monitoring.go

```diff
@@ -91,66 +125,84 @@ func run(
 		case <-ticker.C:
 			if err := pollData(ctx, c, config.Moniker, networkID); err != nil {
 				log.Printf("error polling data for %v: %v", config.Moniker, err)
+			} else {
+				log.Printf("data polled successfully for: %v", config.Moniker)
 			}
 		}
 	}
 }
 
 func pollData(ctx context.Context, client *rpc.Client, moniker, networkID string) error {
+	errs := []error{}
 	if err := sidecarsVersion(ctx, client, moniker, networkID); err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
 	if err := nodeVersion(ctx, client, moniker, networkID); err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
 	if err := latestBlockAndTimestamp(ctx, client, moniker, networkID); err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
-	return nil
+	return errors.Join(errs...)
 }
 
 func latestBlockAndTimestamp(ctx context.Context, client *rpc.Client, moniker, networkID string) error {
 	result := struct {
 		Number    string `json:"number"`
 		Timestamp string `json:"timestamp"`
 	}{}
+
 	err := client.CallContext(ctx, &result, ethGetBlockByNumberEndpoint, "latest", false)
 	if err != nil {
 		mezodUpGauge.WithLabelValues(moniker, networkID).Set(0)
+		// set latest block to 0 for error
+		mezoLatestBlockGauge.WithLabelValues(moniker, networkID).Set(0)
+		// set latest timestamp to 0 for error
+		mezoLatestTimestampGauge.WithLabelValues(moniker, networkID).Set(0)
```

Comment:
I think we are overcomplicating this function. Why not doing the following:
1. We create a variable `err` at the beginning.
2. We create a `defer` that sets all the gauges managed by this fn (latest block and timestamp) to `0` if `err != nil`.
3. Upon an error set `err` and `return`. This way, the defer function will zero the gauges. Of course, we bubble the error up to the caller.

Reply:
addressed in: d1755782d3115b584b67e6fb763dc3d62386ef81